client/modules/workspace/src/SelectModal/SelectModal.tsx

```diff
@@ -0,0 +1,257 @@
+import { Button, Modal, Select } from 'antd';
+import {
+  useAuthenticatedUser,
+  usePathDisplayName,
+  useGetSystems,
+} from '@client/hooks';
+
+import { BaseFileListingBreadcrumb } from '@client/common-components';
+import styles from './SelectModal.module.css';
+import {
+  FileListingTable,
+  TFileListingColumns,
+} from '@client/common-components';
+import { SelectModalProjectListing } from './SelectModalProjectListing';
+
+type TModalChildren = (props: {
+  onClick: React.MouseEventHandler<HTMLElement>;
+}) => React.ReactElement;
+
+const api = 'tapis';
+const HeaderTitle: React.FC<{
+  api: string;
+  system: string;
+  path: string;
+}> = ({ api, system, path }) => {
+  const getPathName = usePathDisplayName();
+  return (
+    <span style={{ fontWeight: 'normal' }}>
+      <i role="none" className="fa fa-folder-o">
+        &nbsp;&nbsp;
+      </i>
+      {getPathName(api, system, path)}
+    </span>
+  );
+};
+
+function getFilesColumns(
+  api: string,
+  system: string,
+  path: string,
+  selectionCallback: (path: string) => void,
+  navCallback: (path: string) => void
+): TFileListingColumns {
+  return [
+    {
+      title: <HeaderTitle api={api} system={system} path={path} />,
+      dataIndex: 'name',
+      ellipsis: true,
+
+      render: (data, record) => (
+        <Button
+          style={{ marginLeft: '3rem', textAlign: 'center' }}
+          onClick={() => navCallback(encodeURIComponent(record.path))}
+          type="link"
+        >
+          <i
+            role="none"
+            style={{ color: '#333333' }}
+            className="fa fa-folder-o"
+          />
+          &nbsp;&nbsp;
+          {data}
+        </Button>
+      ),
+    },
+    {
+      dataIndex: 'path',
+      align: 'end',
+      title: '',
+      render: (_, record) => (
+        <Button
+          type="primary"
+          onClick={() =>
+            selectionCallback(`${api}://${record.system}${record.path}`)
+          }
+        >
+          Select
+        </Button>
+      ),
+    },
+  ];
+}
+
+export const SelectModal: React.FC<{
+  onSelect: (value: string) => void;
+  children: TModalChildren;
+}> = ({ onSelect, children }) => {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const showModal = () => setIsModalOpen(true);
+  const handleClose = () => setIsModalOpen(false);
+  const { user } = useAuthenticatedUser();
+  const {
+    data: { storageSystems, defaultStorageSystem },
+  } = useGetSystems();
+  const includedSystems = storageSystems.filter((s) => s.enabled);
+
+  const systemOptions = includedSystems.map((system) => ({
+    label: system.notes?.label ?? system.id,
+    value: system.id,
+  }));
+  systemOptions.push({ label: 'My Projects', value: 'myprojects' });
+
+  const defaultParams = useMemo(
+    () => ({
+      selectedApi: api,
+      selectedSystem: defaultStorageSystem.id,
+      selectedPath: encodeURIComponent('/' + user?.username),
```

Comment:
Going to Community Data navigates to `Community Data / sal`, which I think is unintended

Reply:
oh, will fix.